patmat/src/main/scala/patmat/Huffman.scala

```diff
@@ -92,7 +108,16 @@ trait Huffman extends HuffmanInterface:
    * If `trees` is a list of less than two elements, that list should be returned
    * unchanged.
    */
-  def combine(trees: List[CodeTree]): List[CodeTree] = ???
+  def combine(trees: List[CodeTree]): List[CodeTree] = {
+    trees match {
+      case x::y::xs =>
```

Comment:
can't it be that `xs` is `Nil`? It could be more efficient not to `span` an empty list.

Reply:
> can't it be that `xs` is `Nil`? It could be more efficient not to `span` an empty list.

spanning an empty list will create 2 empty lists, so "leftOfInsertion ::: (newNode :: rightOfInsertion)" becomes "Nil ::: (newNode :: Nil)" which is fine. I think the performance hit from creation those empty lists and concating them is negligible, but I agree that maybe it's better to handle this case explicitly